### ISSUE TITLE
It takes a "Village" to translate "Hamlet"

### ISSUE THREAD

**Issue by dkurth** [dariusk/NaNoGenMo-2015#18]
This is my first time participating.  I haven't settled on an idea yet, but I hope to finish something.

**Comment by MichaelPaulukonis** [comment]
Welcome! Something is better than nothing. Commit early, iterate often. Fail fast.

**Comment by dkurth** [comment]
I started this morning.  I'm going to extract all dialog from an English text, run it through the Yandex Translator (which is like Google Translate, but their API is free) to translate it to another language, then translate back to English and re-insert it into the original text.  I'll tune the number of translations and which languages I use until I get an interesting result.

**Comment by dkurth** [comment]
I ended up just doing a telephone-game style translation of a Hamlet from English to Japanese to Welsh to Swahili to Yiddish, then back to English.

Code: https://github.com/dkurth/nanogenmo2015
Final product: https://raw.githubusercontent.com/dkurth/nanogenmo2015/master/hamlet.out.txt

The text is impressively preserved, sometimes simply dumbed-down (e.g., the word "husbandry" ends up as "agriculture," and of course there are some funny parts.  
- "Francisco at his post." becomes "Francisco e-mail."  
- The abbreviation "Ber" (for Bernardo) becomes "Eyes fiber." (Not sure why!)  
- "Affection! pooh! you speak like a green girl," becomes "Love it! Fu! Talk like a girl."  
- And, of course, "Hamlet" often becomes "The Village."

This was fun!

**Comment by hugovk** [comment]
"To be, or not to be- that is the question" is clearly too complicated for Hamlet, sorry, I mean for Pork.

> Pork. 
> In the question - 
> Whether tis nobler in the mind. 
> The slings and arrows of outrageous fortune. 
> Armed against a sea of Troubles 
> By your observation? Death. - - 
> Sleep to say we end 
> The heartache and a thousand natural disasters. 
> The meat, there is no choice. 
> Good. Death sleep. 
> . Perchance to dream: - ay friction; 
> And in that sleep of death, what dreams, 
> When we mix death. 
> There's no delay in connection with 
> This tragedy, the long-term; 
> The load on the whips and scorns of time 
> See the evil, the proud contumely, 
> The pain of despis, the laws delay, 
> The insolence of office and the spurns 
> Be patient, the features of the value of 
> If they do the quietus 
> A bare'body kin? These fardels, 
> Groaning and sweat under a weary life, 
> The fear of death? 
> The undiscover country from the bone. 
> The trip back, - in puzzles 
> I'm using an argument that it's bad. 
> Than fly to others that we know? 
> Like the good cowards of all 
> If it's the color resolution 
> Is sicklied of the hall of the pale cast of thought; 
> The company's training and this time, 
> That's what, now on the left. 
> And name.-- Soft! 
> The fair Ophelia!-- Nymph, in thy orisons 
> All my sins remember. 

Hmm, but only 25k (reduced from 32k). Perhaps run it on a larger input?

**Comment by dkurth** [comment]
Yeah, it didn't meat (pun intended) the word count.  I may run it on something longer, but due to the API limitations (1M characters per day), it will take a couple of days to complete a run unless I do fewer languages.  We'll see if I get to that.

**Comment by greg-kennedy** [comment]
Many publications of Shakespeare include the original play on the left page, and the "modern English" translation on the right.  You could put this into a PDF with your translation on the even-numbered pages, so the lines match up, and that would meet the word count : )

**Comment by dkurth** [comment]
Great idea @greg-kennedy, I'll give that a shot!

**Comment by hugovk** [comment]
Good idea! 

Or even interleave lines http://stackoverflow.com/a/4011824/724176 

`paste -d \n\n hamlet.txt hamlet.out.txt > new.txt`

> Ham.
> Pork. 
> To be, or not to be,--that is the question:--
> In the question - 
> Whether 'tis nobler in the mind to suffer
> Whether tis nobler in the mind. 
> The slings and arrows of outrageous fortune
> The slings and arrows of outrageous fortune. 
> Or to take arms against a sea of troubles,
> Armed against a sea of Troubles 
> And by opposing end them?--To die,--to sleep,--
> By your observation? Death. - - 
> No more; and by a sleep to say we end
> Sleep to say we end 
> The heartache, and the thousand natural shocks
> The heartache and a thousand natural disasters. 

Which would look better with different formatting, but Greg's idea will probably better overall.

**Comment by dkurth** [comment]
In great haste, I hacked together a side-by-side version here:

https://github.com/dkurth/nanogenmo2015/blob/master/hamlet.sidebyside.pdf

Thanks for the great suggestions!  Looks like the final word count is 56,765.

**Comment by hugovk** [comment]
Nice! Congrats! Have a completed label!

---

> **Ham.**
> I humbly thank you; well, well, well. 

=>

> **Pork.**
>  I humbly Thank you, etc
